mlir/include/mlir-c/IR.h

```diff
@@ -552,25 +552,69 @@ MLIR_CAPI_EXPORTED intptr_t mlirOperationGetNumSuccessors(MlirOperation op);
 MLIR_CAPI_EXPORTED MlirBlock mlirOperationGetSuccessor(MlirOperation op,
                                                        intptr_t pos);
 
+/// Returns an inherent attribute attached to the operation given its name.
```

Comment:
Nit: what happens when there is no attribute? (can we differentiate between an optional inherent attribute being absent and the name not corresponding to an inherent attribute?)

Reply:
We could, I wasn't sure if it was useful for the C API, the alternative may be to add a `bool hasInherentAttr(name)` API that the use could check on top of this one to distinguish?

Reply:
(I just added this API)

Reply:
I just added this API, but let me know if you have a better idea.